src/plugins/intel_cpu/src/nodes/unique.cpp

```diff
@@ -0,0 +1,399 @@
+        if (axis < 0) {
+            axis += op->get_input_partial_shape(IN_DATA).rank().get_length();
+        }
+        if (axis < 0 || axis >= op->get_input_partial_shape(IN_DATA).rank().get_length()) {
+            THROW_ERROR << "has invalid axis value: " << ov::as_type<ov::op::v0::Constant>(op->get_input_node_ptr(AXIS))->cast_vector<int>()[0];
+        }
+    } else {
+        flattened = true;
+    }
+}
+
+void Unique::initSupportedPrimitiveDescriptors() {
+    dataPrecision = getOriginalInputPrecisionAtPort(IN_DATA);
+    if (dataPrecision != Precision::I32 && dataPrecision != Precision::I8 && dataPrecision != Precision::U8) {
+        dataPrecision = Precision::FP32;
+    }
+    dataTypeSize = dataPrecision.size();
+    const InferenceEngine::Precision axisPrecision = Precision::I32;
+
+    impl_desc_type implType = ref;
+
+    std::vector<PortConfigurator> inPortConfigs = { {LayoutType::ncsp, dataPrecision} };
+    if (!flattened) {
+        inPortConfigs.push_back({LayoutType::ncsp, axisPrecision});
+    }
+    std::vector<PortConfigurator> outPortConfigs;
+    for (int i = 0; i < 4; i++) {
+        outPortConfigs.push_back({LayoutType::ncsp, i == 0 ? dataPrecision : axisPrecision});
+    }
+
+    addSupportedPrimDesc(inPortConfigs, outPortConfigs, implType, isDynamicNode());
+}
+
+void Unique::createPrimitive() {
+    Node::createPrimitive();
+}
+
+void Unique::prepareParams() {
+    auto& dataMemPtr = getParentEdgeAt(IN_DATA)->getMemoryPtr();
+    if (!dataMemPtr || !dataMemPtr->isAllocated()) {
+        THROW_ERROR << " has not allocated input data memory.";
+    }
+    for (int i = 0; i < 4; i++) {
+        if (definedOutputs[i]) {
+            auto& dstMemPtr = getChildEdgeAt(i)->getMemoryPtr();
+            if (!dstMemPtr || !dstMemPtr->isAllocated()) {
+                THROW_ERROR << " has not allocated output memory at port " << i;
+            }
+        }
+    }
+    if (getSelectedPrimitiveDescriptor() == nullptr) {
+        THROW_ERROR << " has unidentified preferable primitive descriptor.";
+    }
+}
+
+void Unique::execute(dnnl::stream strm) {
+    if (flattened) {
+        size_t uniqueLen = 1lu;
+        if (dataPrecision == Precision::FP32) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::FP32>::value_type>();
+        } else if (dataPrecision == Precision::I32) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::I32>::value_type>();
+        } else if (dataPrecision == Precision::I8) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::I8>::value_type>();
+        } else if (dataPrecision == Precision::U8) {
+            uniqueLen = flattenTensorExec<PrecisionTrait<Precision::U8>::value_type>();
+        }
+
+        const size_t srcLen = getParentEdgeAt(IN_DATA)->getMemoryPtr()->GetSize() / dataPrecision.size();
+        redefineOutputMemory({ {uniqueLen}, {uniqueLen}, {srcLen}, {uniqueLen}});
```

Comment:
seems like redefineOutputMemory() happens after all computation finished, which means the computation result stored in orignal child edge is lost. Is there any copy for the computation result to this new allocated memory?

Reply:
1. ok, maybe a special case need double check. Do we support dynamic input shape for this op? If yes and initial shape is undefined(low/upper bound is not provide), then there is possibility that real reallocation happens. for example, first infer input value is [1, 1, 1], second infer input value is [1,2,3,4,5]. second uniqueLen is even larger than first srcLen(preallocate with this size), which need reallocation. Am I right?
2. I see when shape is the same, there will be no reallocation in https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/node.cpp#L560-L561. could you please help me understand where the reallocation will be skipped when new shape is smaller than before.

Reply:
1. That is right point for dynamic shapes. I lost this initialization.
2. https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/cpu_memory.cpp#L185

Reply:
That's completely unsafe way of memory redefinition. You are exploiting knowledge of current memory reallocation behavior, which in fact depends on underlaying memory manager. Actual memory manager - not under Node control, so it's change can make this logic invalid. 
So @chenhu-wang  is right: you need to use temporary memory object and then copy data to dst.

Reply:
Fixed